docs/design/rosbag2_record_replay_service.md

```diff
@@ -0,0 +1,73 @@
+# Rosbag2 record and replay service
+
+## Context
+
+In [ros2/rosbag2#773](https://github.com/ros2/rosbag2/issues/773), it was recommended that record service requests feature should be added to rosbag2.  
+
+This design describe how to implement this feature based on current architecture and how to extend current command parameter.  
+
+## How to save record service message
+
+Service introspection ([ros2/ros2#1285](https://github.com/ros2/ros2/issues/1285)) had been implemented. All service messages (requests and responses) are published to service introspection topic which is hidden topic named as `_service_event` under service name (e.g. /add_two_ints/_service_event). So all messages in this topic can be recorded like general topic. 
+
+## How to distinguish service introspection topic in all recorded topics 
+
+Service introspection is saved like general topic. But they should use different process after reading from recording file. So they should be distinguished by topic name.  
+
+- General topic  
+  
+  [NameSpace/]TopicName
+
+- Hidden topic  
+  
+  [NameSpace/]_TopicName
+
+- Service introspection topic
+  
+  [NameSpace/]ServiceName/_service_event
```

Comment:
if we specify `ros2 bag record --include-hidden-topics`, this will be included as hidden topic currently? if that is so, `/_service_event` should be excluded from topic category for rosbag2 by default?

Reply:
Yes. If the user doesn't want to record service, service event topic ([NameSpace/]ServiceName/_service_event) is ignored even if the user set `--include-hidden-topics`.

Reply:
> IMO it would be more consistent to include service events in --include-hidden-topics since it is, after all, a hidden topic and not something special

In `rosbag2 info`, service information will be listed separately. So I prefer not to categorize service event topics as general hidden topics.  
What do you think ?

Reply:
Currently, we have some special options for service, such as `--all-services, --services`. So whether service should not be affected by `--include-hidden-topics`.  That is, service record cannot be enabled by `--include-hidden-topics`.  Users must explicitly indicate their intention to record services via e.g. `--all-services, --services`.   
What do you think ? @MichaelOrlov @ihasdapie

Reply:
I tend to agree with @ihasdapie 
> IMO it would be more consistent to include service events in --include-hidden-topics since it is, after all, a hidden topic and not something special

It is what it is. Service events are hidden topics.

Reply:
As far as I recall we have made the decision about `--include-hidden-topics` at one of the `Rosbag2 and Tooling WG` meetings last month. 
> 2. Decided that --include-hidden-topics will record service events as well, since it is just a hidden topic.

Here is the full summary from the meeting https://github.com/ros2/rosbag2/pull/1359#issuecomment-1574575492

Reply:
@MichaelOrlov Thanks for your confirmation. 

> As far as I recall we have made the decision about --include-hidden-topics at one of the Rosbag2 and Tooling WG meetings last month.

Yes. It is because new record options `--all-services` and `--all-topics` were added later. There is an option used to enable recording services. Now `--include-hidden-topics` also has the same effect. If users doesn't know service introspection, maybe they confuse the behavior of `--include-hidden-topics`.   
Anyway, `--include-hidden-topics` will enable recording services.